src/python/pants/base/hash_utils.py

```diff
@@ -16,6 +18,7 @@ def hash_all(strs, digest=None):
   """
   digest = digest or hashlib.sha1()
   for s in strs:
+    s = ensure_binary(s)
```

Comment:
Hasher requires byte strings. Most of the changes made below are converting to bytes.